jwala-services/src/main/java/com/cerner/jwala/service/jvm/impl/JvmServiceImpl.java

```diff
@@ -622,7 +638,15 @@ private void secureCopyJvmConfigJar(Jvm jvm, String jvmConfigJar, User user) thr
 
     private void deployJvmConfigJar(Jvm jvm, User user, String jvmJar) throws CommandFailureException {
         final String parentDir = ApplicationProperties.get("remote.paths.instances");
-        CommandOutput execData = jvmControlService.controlJvm(
+        CommandOutput execData = jvmControlService.executeCreateDirectoryCommand(jvm, parentDir+"/"+jvm.getJvmName());
+        if (execData.getReturnCode().wasSuccessful()) {
+            LOGGER.info("Successfully created the parent directory {}", parentDir);
+        } else {
+            String standardError = execData.getStandardError().isEmpty() ? execData.getStandardOutput() : execData.getStandardError();
+            LOGGER.error("Deploy command completed with error trying to extract and back up JVM config {} :: ERROR: {}", jvm.getJvmName(), standardError);
+            throw new InternalErrorException(FaultType.REMOTE_COMMAND_FAILURE, standardError.isEmpty() ? CommandOutputReturnCode.fromReturnCode(execData.getReturnCode().getReturnCode()).getDesc() : standardError);

```

Comment:
I know InternalErrorException has been used in this application for some time now, I'm not in favor of it, my 2 cents is to throw an service exception that relates to this class. The reason is when you see the exception, you would immediately see InternalErrorException - it does not tell you right away where the problem happened.